### PR TITLE
downgrade the System.Text.Json package reference version

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

*This pull request fixes issue #1878.*

### Description

*Briefly describe the changes of this pull request.*

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Review by habbes — APPROVED** [review]

**Review by marabooy — APPROVED** [review]

File: src/Microsoft.OData.Edm/Microsoft.OData.Edm.csproj [hunk 1]
```diff
@@ -94,9 +94,7 @@
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
-    <PackageReference Include="System.Text.Json">
-      <Version>4.7.1</Version>
-    </PackageReference>
+    <PackageReference Include="System.Text.Json" Version="4.*" />
```
Comment (gathogojr) [inline src/Microsoft.OData.Edm/Microsoft.OData.Edm.csproj:97]: @xuzhg Curious how `Version=4.*` is interpreted. Wouldn't 4.7.1 be validated against 4.*?

Comment (xuzhg) [reply src/Microsoft.OData.Edm/Microsoft.OData.Edm.csproj:97]: Here's the context:

Floating versions
A floating dependency version is specified with the * character. For example, 6.0.*. This version specification says "use the latest 6.0.x version"; 4.* means "use the latest 4.x version." Using a floating version reduces changes to the project file, while keeping up to date with the latest version of a dependency.

When using a floating version, NuGet resolves the highest version of a package that matches the version pattern, for example 6.0.* gets the highest version of a package that starts with 6.0:

Choosing version 6.0.1 when a floating version 6.0.* is requested

Comment (xuzhg) [reply src/Microsoft.OData.Edm/Microsoft.OData.Edm.csproj:97]: @gathogojr Besides, I use this PR and output a nightly, That nightly can be used directly without the build error.

Comment (gathogojr) [reply src/Microsoft.OData.Edm/Microsoft.OData.Edm.csproj:97]: @xuzhg Got it. So having the version pinned to 4.7.1 was the issue

**Review by gathogojr — APPROVED** [review]

**Comment by gathogojr** [comment]
@xuzhg Any idea why ADO is reporting building failure for [OData-master-pipeline-Rolling] in spite of this:
image

**Review by habbes — APPROVED** [review]

**Comment by habbes** [comment]
@xuzhg are the builds failing as a result of this change?

**Comment by xuzhg** [comment]
> @xuzhg are the builds failing as a result of this change?

No, i run the rolling manually, it can pass. See https://identitydivision.visualstudio.com/OData/_build/results?buildId=621841&view=results

**Pull request merged**